variables.tf

```diff
@@ -4,6 +4,36 @@ variable "use_msi_to_authenticate" {
   default     = false
 }
 
+# 3 MSI configuration methods:
+   # System Assigned Managed Identity: supply Object ID
+   # User Assigned Managed Identity  : set use_user_assigned_msi, supply name/resource group; Sources objectID at runtime
+   # User Assigned Managed Identity  : set use_user_assigned_msi, supply objectId directly
+variable "msi" {
+  description = "MSI configuration information."
+  type        = object({
+    use_user_assigned_msi = bool
+
+    name                  = optional(string) 
+    resource_group_name   = optional(string)
+
+    object_id             = optional(string)
+  })
+
+  default = {
```

Comment:
Botched it. I don't want this default value. One sec.

Reply:
Corrected this to default to 'null'. 

The logic hinges on the msi block's presence to produce the functionality of "use_msi_to_authenticate", so it can't default to an object being there (or it defaults to always trying to use msi).